cmake/version.cmake

```diff
@@ -1,5 +1,33 @@
 # SPDX-License-Identifier: Apache-2.0
 
+#.rst:
+# version.cmake
+# -------------
+#
+# Inputs:
+#
+#   ``*VERSION*`` and other constants set by
+#   maintainers in ``${ZEPHYR_BASE}/VERSION``
+#
+# Outputs with examples::
+#
+#   PROJECT_VERSION          1.14.99.42
```

Comment:
If the idea was "to explain everything", then I'd like to point that the list below doesn't explain what is "42" at the end.

Reply:
Despite the "42" example (changed, thx) the goal is to explain but not "everything". This is more about not having to read the whole cmake file or git grep the entire source code to filter out the many temporary variables and find the names that matters. The added comment points at the VERSION file where TWEAK is defined so this is hopefully enough. I considered mentioning TWEAK here but @sebastianboe just helped me trim this down significantly (thx) so I'm reluctant to add stuff again.